memo/src/instruction.rs

```diff
@@ -0,0 +1,49 @@
+use crate::error::MemoError;
+use solana_sdk::{entrypoint::ProgramResult, info, program_error::ProgramError};
+use std::{mem::size_of, str::from_utf8};
+
+/// Instructions supported by the memo program
+#[repr(C)]
+#[derive(Clone, Debug, PartialEq)]
```

Comment:
Why the handwritten serialize/deserialize? Seems like Serde and Bincode would work fine.

Reply:
Yeah, it could. I followed the Token program's lead on this to shrink the instruction size to 1 byte. Serde would make it 4 bytes.